src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -86,7 +108,22 @@ var DOMRenderer = ReactFiberReconciler({
     hostContext : string | null,
     internalInstanceHandle : Object,
   ) : Instance {
-    const domElement : Instance = createElement(type, props, rootContainerInstance, hostContext);
+    let parentNamespace;
+    if (__DEV__) {
+      parentNamespace = hostContext ? hostContext.namespace : null;
+      if (hostContext) {
+        validateDOMNesting(type, null, internalInstanceHandle, hostContext.ancestorInfo,);
+      }
+      if (typeof props.children === 'string' || typeof props.children === 'number') {
```

Comment:
No because this is the "single child" case in which we don't create an instance but use `setTextContent`.